{{cookiecutter.project_name}}/noxfile.py

```diff
@@ -1,20 +1,67 @@
 import nox
+from pathlib import Path
+import os
 
 # Options to modify nox behaviour
+nox.options.default_venv_backend = "uv|virtualenv"
 nox.options.reuse_existing_virtualenvs = True
 
+VENV_DIR = Path("./venv").resolve()
+
 @nox.session(name="docs")
-def build_docs(session):
+def build_docs(session: nox.Session) -> None:
     """Build the documentation and load it in a browser tab, rebuilding on changes."""
-    envbindir = session.bin
     session.install("-e", ".[docs]")
     with session.chdir("docs/"):
-        session.run(
-            "sphinx-autobuild",
-            "-j",
-            "auto",
-            "--open-browser",
-            "-qT",
-            ".",
-            f"{envbindir}/../tmp/html",
-        )
+        # For local development
+        if session.interactive:
+            session.run(
+                "sphinx-autobuild",
+                "-j",
+                "auto",
+                "--open-browser",
+                "-qT",
+                ".",
+                "build/html/",
+            )
+        # For CI testing if documentation builds
+        else:
+            session.run(
+                "sphinx-build",
+                "-b",
+                "html",
+                "-W",
+                "--keep-going",
+                ".",
+                "build/html/",
+            )
+
+@nox.session(name="generated-project-tests")
+def run_generated_project_tests(session):
+    """Run the tests for testing units inside generated project"""
+    session.install("setuptools", silent=False)
+    session.install("-e", ".[dev]", silent=False)
+    session.run("pytest", "tests/generated_project_tests")
+
+@nox.session(name="user-tests")
+def run_user_tests(session):
+    """Run the tests for testing user written tests"""
+    session.install("setuptools", silent=False)
+    session.install("-e", ".[dev]", silent=False)
+    session.run("pytest", "tests/user_tests")
+
+@nox.session(name="coverage")
+def run_coverage(session):
+    """Run the coverage tests and generate an XML report."""
+    session.install("setuptools", silent=False)
+    session.install("coverage", silent=False)
+    session.install("-e", ".[all,dev,jax]", silent=False)
+    session.run("pytest", "--cov=src/pybamm_cookiecutter", "--cov-report=xml", "tests/")
```

Comment:
Cleaner. `coverage` should be installed as a `dev` dependency if it is not being installed in that group at the moment.
```suggestion
    session.posargs.append("--cov=src/pybamm_cookiecutter")
    session.posargs.append("--cov-report=xml")
    run_user_tests(session)
```